src/lerna/changelog/index.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { handleErrorMessage, handleSuccessMessage } from '../../utils';
+import { Changelog } from 'lerna-changelog';
+import { Command } from 'commander';
+import { Ora } from 'ora';
+import execa from 'execa';
+import { fromPath } from 'lerna-changelog/lib/configuration';
+import { token as getToken } from '../../github';
+
+type ARGS = {
+  from?: string;
+  to?: string;
+  path?: string;
+  token?: string;
+  spinner?: Ora;
+};
+
+/**
+ * Execute the `lerna-changelog` command.
+ *
+ * @param {string} [args.from] GitHub ref (commit SHA, tag).
+ * @param {string} [args.to] GitHub ref (commit SHA, tag).
+ * @param {string} [args.path] Path to a git directory.
+ * @param {string} [args.token] GitHub personal access token.
+ * @param {Ora} [args.spinner] Terminal spinner.
+ *
+ * @returns {Promise<string>} The markdown result of the changelog command.
+ */
+export const execute = async (args: ARGS = {}): Promise<string | undefined> => {
+  let retVal: string | undefined;
+
+  try {
+    process.env.GITHUB_AUTH = args.token || (await getToken(args.spinner));
+
+    const revParseArgs = ['rev-parse', '--show-toplevel'];
+    const describeArgs = ['describe', '--abbrev=0', '--tags'];
+
+    if (args.path) {
+      revParseArgs.unshift('-C', args.path);
+      describeArgs.unshift('-C', args.path);
+    }
+
+    const rootPath = await execa('git', revParseArgs);
+    const config = fromPath(rootPath.stdout);
+    const changelog = new Changelog(config);
+    let tagTo = args.to;
+    let tagFrom = args.from;
```

Comment:
This tag generation logic is heavily specialized for the `react-components` repo? Is this a sane default for other repos that would use this?

Reply:
Yes, I think the [documented defaults](https://github.com/zendeskgarden/tool-shed/tree/jzempel/init/src/lerna/changelog#readme) are by far the typical case. Obviously both `from` and `to` can be overriden.